src/idl_gen_rust.cpp

```diff
@@ -1425,11 +1423,7 @@ class RustGenerator : public BaseGenerator {
       case ftVectorOfBool:
       case ftVectorOfFloat: {
         const auto typname = GetTypeBasic(type.VectorType());
-        const auto vector_type =
-            IsOneByte(type.VectorType().base_type)
-                ? "&" + lifetime + " [" + typname + "]"
-                : "flatbuffers::Vector<" + lifetime + ", " + typname + ">";
-        return WrapOption(vector_type);
+        return WrapOption("flatbuffers::Vector<" + lifetime + ", " + typname + ">");
```

Comment:
This is the only major change in this PR that I expect users to notice, previously calling the table accessor on structs, or primitives would return a slice. It now returns `Vector`.

We could keep the behaviour for structs and bytes, using safe_slice, but I felt this was just more confusing.

Reply:
I agree, I think sometimes using slice and sometimes using flatbuffers::Vector is confusing for just about everyone.